tests/test_pytypes.py

```diff
@@ -133,6 +133,14 @@ def __repr__(self):
     else:
         assert m.str_from_handle(malformed_utf8) == "b'\\x80'"
 
+    assert m.str_from_string_from_str("this is a str") == "this is a str"
+    ucs_surrogates_str = "\udcc3"
+    if env.PY2:
+        assert u"\udcc3" == m.str_from_string_from_str(ucs_surrogates_str)
```

Comment:
Howdy Shane, thanks!
Any chance you know what's going wrong in the Python 2.x code?
```
UnicodeEncodeError: 'ascii' codec can't encode character u'\udcc3' in position 38: ordinal not in range(128)
```
https://github.com/pybind/pybind11/pull/2903/checks?check_run_id=2115679652#step:11:98

I can take a deeper look if ya want.